dropbox/dropbox_client.py

```diff
@@ -675,20 +709,59 @@ def __exit__(self, *args):
         self.close()
 
 
+def _check_user_team_auth_args(oauth2_access_token,
+                               oauth2_refresh_token,
+                               app_key):
+    if not (oauth2_access_token or oauth2_refresh_token):
+        raise BadInputException('OAuth2 access token or refresh token must be set')
+    if oauth2_refresh_token and not app_key:
+        raise BadInputException("app_key is required to refresh tokens")
+
+
 class Dropbox(_DropboxTransport, DropboxBase):
     """
     Use this class to make requests to the Dropbox API using a user's access
     token. Methods of this class are meant to act on the corresponding user's
     Dropbox.
     """
-    pass
+    def _check_auth_args(self,
+                         oauth2_access_token,
+                         oauth2_refresh_token,
+                         oauth2_access_token_expiration,
+                         app_key,
+                         app_secret):
+        _check_user_team_auth_args(oauth2_access_token, oauth2_refresh_token, app_key)
+
+
+class DropboxAppAuth(_DropboxTransport, DropboxBase):
```

Comment:
Why is this class necessary? It seems the only difference is just setting this check_auth_args function. I think generally we want to unify the constructors into a single multi-use constructor rather than making additional specific constructors

Reply:
I added this because in some cases it might make sense to skip requirements other than app key and secret when developers only need to use app auth routes. If we don't have an app auth client class then they have to provide dummy fields for oauth tokens or refresh tokens. But it makes sense to have a unified constructor as well